sdk/core/core-https/documentation/core2.md

```diff
@@ -0,0 +1,69 @@
+
+The downside is that much of the code had evolved organically over time, experienced incomplete upgrades to various layers, and had dead code that wasn't really needed anymore. We mitigated this during the 1.0 GA timeframe by manually declaring cleaner public interfaces (e.g. `OperationOptions` over `RequestOptionsBase`) but many of the internal interfaces were still messy, loosely typed, or confusing.
+
+As the JS team has gained experience shipping track 2 client libraries, we realized that the technical debt accumulated in `@azure/core-http` was costing us in terms of maintainability and performance. After multiple internal discussions about what exactly needed to change in order to move forward, a plan emerged to modernize `@azure/core-http` and give us a more robust core to build upon.
+
+### Important goals and considerations:
+
+- The request pipeline should be re-usable outside of AutoRest generated code (e.g. Cosmos.)
+- The request pipeline should be easily customizable by client libraries and consumers.
+- Performance of the majority use case should be optimized and have few allocations.
+- Future improvements to codegen should be easy to phase in.
+
+## Scope of work
+
+### Separation of Concerns
+
+One major goal of this work is to decompose the library into three separate packages: `@azure/core-https`, `@azure/core-client`, and `@azure/core-xml`. This is to enable smaller dependency graphs and bundles for individual client libraries.
+
+The first package, `@azure/core-https` will not be AutoRest-specific and is intended to be usable by any package that wants to make requests in a way that follows our SDK design guidelines.
+
+`@azure/core-client` is intended to be a runtime dependency of any AutoRest-generated client library and has all necessary helpers except for XML (de)serialization. Packages that require XML support will additionally depend upon `@azure/core-xml`.
+
+### Pipeline improvements
+
+A major weakness of the existing JS Core was that the request pipeline system was difficult to customize at the library level. This placed an undue burden on authors of libraries and increased support costs. As other languages had better support for customizing the request pipeline, we had a desire to provide a simple and flexible way to add and order policies that could affect both HTTP requests and responses.
+
+The [New Pipeline Design](https://github.com/Azure/azure-sdk-for-js/issues/8461) solves these concerns by providing a new abstraction that is able to reliably apply ordered policies with minimal configuration.
+
+### Dropping legacy browser support
+
+As 2.0 is intended to be a modernization effort, it will **not** support Internet Explorer. This does mean that client libraries which need to support IE will not be able to upgrade until such time as they are not required to support legacy browsers.
```

Comment:
A line on what we have to gain by dropping the support would be nice

Reply:
Added another line for this.